# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1411330372 14400
#      Sun Sep 21 16:12:52 2014 -0400
# Node ID faa6a98f8d4bf5c5797ee20dab903d87c57f7e13
# Parent  b8bd0b55af45883fa7e6df46ce5716afe943335a
fix crash in kron introduced in changeset aa9ca67f09fb

* kron.cc (kron (const PermMatrix& a, const PermMatrix& b)):
Allocate na*nb elements for temporary array.

diff --git a/libinterp/corefcn/kron.cc b/libinterp/corefcn/kron.cc
--- a/libinterp/corefcn/kron.cc
+++ b/libinterp/corefcn/kron.cc
@@ -133,17 +133,17 @@ kron (const MSparse<T>& A, const MSparse
 
 static PermMatrix
 kron (const PermMatrix& a, const PermMatrix& b)
 {
   octave_idx_type na = a.rows ();
   octave_idx_type nb = b.rows ();
   const Array<octave_idx_type>& pa = a.col_perm_vec ();
   const Array<octave_idx_type>& pb = b.col_perm_vec ();
-  Array<octave_idx_type> res_perm;
+  Array<octave_idx_type> res_perm (na * nb);
   octave_idx_type rescol = 0;
   for (octave_idx_type i = 0; i < na; i++)
     {
       octave_idx_type a_add = pa(i) * nb;
       for (octave_idx_type j = 0; j < nb; j++)
         res_perm.xelem (rescol++) = a_add + pb(j);
     }
 
